specification/trace/semantic_conventions/instrumentation/elasticsearch.md

```diff
@@ -0,0 +1,50 @@
+# Semantic conventions for Elasticsearch
+
+**Status**: [Experimental](../../../document-status.md)
+
+This document defines semantic conventions to apply when creating a span for requests to Elasticsearch.
+
+## Span Name
+
+The **span name** SHOULD be of the format `<endpoint id>`.
+
+The elasticsearch endpoint identifier is used instead of the url path in order to reduce the cardinality of the span 
+name, as the path could contain dynamic values. The endpoint id is the `name` field in the
+[elasticsearch schema](https://raw.githubusercontent.com/elastic/elasticsearch-specification/main/output/schema/schema.json).
+If the endpoint id is not available, the span name SHOULD be `http.request.method`.
+
+## URL path parts
+
+Many Elasticsearch url paths allow dynamic values. These SHOULD be recorded in span attributes in the format
+`db.elasticsearch.path_parts.<key>`, where `<key>` is the url path part name. The implementation SHOULD
+reference the [elasticsearch schema](https://raw.githubusercontent.com/elastic/elasticsearch-specification/main/output/schema/schema.json)
+in order to map the path part values to their names.
+
+| Attribute                           | Type | Description                           | Examples                                                                                 | Requirement Level |
+|-------------------------------------|---|---------------------------------------|------------------------------------------------------------------------------------------|---|
+| `db.elasticsearch.path_parts.<key>` | string | A dynamic value in the url path.      | `db.elasticsearch.path_parts.index=test-index`; `db.elasticsearch.path_parts.doc_id=123` | Conditionally Required: [1] |
+
+**[1]:** when the url has dynamic values
+
+## Span attributes
+
+<!-- semconv db.elasticsearch -->
+| Attribute  | Type | Description  | Examples  | Requirement Level |
+|---|---|---|---|---|
+| [`db.operation`](../database.md) | string | The endpoint identifier for the request. [1] | `search`; `ml.close_job`; `cat.aliases` | Required |
+| [`db.statement`](../database.md) | string | The request body for a [search-type query](https://www.elastic.co/guide/en/elasticsearch/reference/current/search.html), as a json string. [2] | `"{\"name\":\"TestUser\",\"password\":\"REDACTED\"}"` | Recommended: [3] |
```

Comment:
This is an interesting conundrum here. There's an OTEP around providing generic "request body" and "response body" attributes on spans.

I think what you've defined here makes a lot of sense in the context of Elastic Search.

Could you provide more guidance on sanitizing sensitive information?   Is it "obvious" what could be sensitive in elastic search from the instrumentation author of the client library, or does this require knowledge of the datamodel in the database?  IF the later, maybe there needs to be recommended configuration options for this attribute.

Reply:
@jsuereth, the request body for an Elasticsearch query could indeed contain sensitive information. Do semantic conventions typically suggest a config option? The Graphql and database `db.statement` fields suggest sanitization but don't mention config options so I'm happy to add something but not sure what type of suggestion.
Graphql: [`The value may be sanitized to exclude sensitive information.`](https://github.com/open-telemetry/semantic-conventions/blob/main/specification/trace/semantic_conventions/instrumentation/graphql.md?plain=1#L20)
database: [`Should be collected by default only if there is sanitization that excludes sensitive information.`](https://github.com/open-telemetry/semantic-conventions/blob/main/specification/trace/semantic_conventions/database.md?plain=1#L187C10-L187C107)

Reply:
As discussed, once your notice changes to the database variant, then this PR looks good to go!

Reply:
done! https://github.com/open-telemetry/semantic-conventions/pull/23/commits/1bd79a1cb4aa851df39e421c9ab13a2cfe2e3952